client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -39,12 +39,21 @@ import {
 
 import { registerHandlers } from 'state/data-layer/handler-registry';
 import getNextPageQuery from 'state/selectors/get-next-page-query';
+import { isEditorIframeLoaded, getEditorIframePort } from 'calypso/state/editor/selectors';
 
 /**
  * Module variables
  */
 const log = debug( 'calypso:middleware-media' );
 
+const createMediaGutenframePayload = ( status, mediaItem ) => ( {
```

Comment:
It seems like this is something that should live outside of the data layer. `lib/media/utils`? Could potentially also be usable in `updateImageBlocks()` in the calypsoify iframe.

Reply:
Should be covered by 28ccf690cba9a2f11b85a9e43c27e31df60d9045 – let me know if that works for you. I removed the `updateImageBlocks` method in `calypsoify-iframe.tsx` because it's not used anymore as we call `iframePort.postMessage` in that newly created `gutenframeUpdateImageBlocks` thunk.